optuna/_optimize.py

```diff
@@ -185,8 +187,58 @@ def _run_trial(
     func: "optuna.study.ObjectiveFuncType",
     catch: Tuple[Type[Exception], ...],
 ) -> trial_module.Trial:
+
     trial = study._ask()
+    return asyncio.run(_return_coroutines(study, trial, func, catch))
```

Comment:
`asyncio.run` is not available in Python 3.6. How about using `loop = ...; loop.run_until_complete`?